--- a/build/core/tasks/kernel.mk
+++ b/build/core/tasks/kernel.mk
@@ -146,6 +146,7 @@
 
 .PHONY: $(TARGET_OUT_UBUNTU_KERNEL)/vmlinuz
 $(TARGET_OUT_UBUNTU_KERNEL)/vmlinuz: $(TARGET_OUT_UBUNTU_KERNEL) $(KERNEL_HEADERS_INSTALL)
+ifeq (,$(wildcard $(OUT_DIR)/../repo/$(TARGET_KERNEL_UBUNTU_SERIES)/linux-image-*-$(TARGET_DEVICE)*.deb))
 	$(hide) $(PULL_LP_BIN) $(TARGET_KERNEL_UBUNTU_META) -o $(TARGET_OUT_UBUNTU_KERNEL) $(TARGET_KERNEL_UBUNTU_SERIES)
 	$(hide) IFS=", "; for dep in \
 		`dpkg-deb -f $(TARGET_OUT_UBUNTU_KERNEL)/$(TARGET_KERNEL_UBUNTU_META)_*.deb Depends`; do \
@@ -155,16 +156,19 @@
 		done; \
 		if [ -n "$$kernel_image" ]; then \
 			$(PULL_LP_BIN) $$kernel_image -o $(TARGET_OUT_UBUNTU_KERNEL) $(TARGET_KERNEL_UBUNTU_SERIES); \
-			dpkg-deb -x $(TARGET_OUT_UBUNTU_KERNEL)/linux-image-[0-9]*.deb $(TARGET_OUT_UBUNTU_KERNEL); \
-			kernel_version=$${kernel_image#linux-image-}; \
-			cp -v $(TARGET_OUT_UBUNTU_KERNEL)/boot/vmlinuz-$$kernel_version $(TARGET_OUT_UBUNTU_KERNEL)/vmlinuz; \
-			cp -a $(TARGET_OUT_UBUNTU_KERNEL)/lib/modules/$$kernel_version $(KERNEL_MODULES_OUT); \
-			depmod -a -b $(TARGET_OUT) $$kernel_version; \
 		else \
 			echo -n "Unable to find a valid linux-image dependency from "; \
 			echo "the meta package $(TARGET_KERNEL_UBUNTU_META), aborting."; \
 			exit 1; \
 		fi;
+else
+	$(hide) cp $(OUT_DIR)/../repo/$(TARGET_KERNEL_UBUNTU_SERIES)/linux-image-*-$(TARGET_DEVICE)*.deb $(TARGET_OUT_UBUNTU_KERNEL)
+endif
+	$(hide) dpkg-deb -x $(TARGET_OUT_UBUNTU_KERNEL)/linux-image-[0-9]*.deb $(TARGET_OUT_UBUNTU_KERNEL); \
+	kernel_version=`dpkg-deb -I $(TARGET_OUT_UBUNTU_KERNEL)/linux-image-[0-9]*.deb | sed -n 's/ Package: linux-image-//p'`; \
+	cp -v $(TARGET_OUT_UBUNTU_KERNEL)/boot/vmlinuz-$$kernel_version $(TARGET_OUT_UBUNTU_KERNEL)/vmlinuz; \
+	cp -a $(TARGET_OUT_UBUNTU_KERNEL)/lib/modules/$$kernel_version $(KERNEL_MODULES_OUT); \
+	depmod -a -b $(TARGET_OUT) $$kernel_version
 
 else ifeq ($(FULL_KERNEL_BUILD),true)
 
--- a/build/core/Makefile
+++ b/build/core/Makefile
@@ -403,7 +403,11 @@
 	$(call pretty,"Target Ubuntu Touch ram disk: $@")
 	$(hide) rm -rf $(TARGET_OUT_UBUNTU_INITRD)
 	$(hide) mkdir -p $(TARGET_OUT_UBUNTU_INITRD)
+ifeq (,$(wildcard $(OUT_DIR)/../repo/$(TARGET_KERNEL_UBUNTU_SERIES)/$(TARGET_UBUNTU_INITRD_PKG)*deb))
 	$(hide) $(PULL_LP_BIN) $(TARGET_UBUNTU_INITRD_PKG) -o $(TARGET_OUT_UBUNTU_INITRD)
+else
+	$(hide) $(ACP) $(wildcard $(OUT_DIR)/../repo/$(TARGET_KERNEL_UBUNTU_SERIES)/$(TARGET_UBUNTU_INITRD_PKG)*deb) $(TARGET_OUT_UBUNTU_INITRD)
+endif
 	$(hide) dpkg-deb -x $(TARGET_OUT_UBUNTU_INITRD)/$(TARGET_UBUNTU_INITRD_PKG)*.deb $(TARGET_OUT_UBUNTU_INITRD)
 	$(hide) $(ACP) $(TARGET_UBUNTU_INITRD) $@
 
